e2etests/test/method_disconnect.js

```diff
@@ -211,20 +208,35 @@ protocolAndTermination.forEach( function (testConfiguration) {
       };
       deviceClient.on('disconnect', disconnectHandler);
       var firstPayload = { k1: uuid.v4() };
-      setMethodHandler(methodName1, firstPayload);
-      sendMethodCall(serviceClient, methodName1, firstPayload, function(err) {
-        if (!err) {
-          debug('got result from first method invocation.  Having the client send the kill');
-          firstMethodSent = true;
-          var terminateMessage = new Message('');
-          terminateMessage.properties.add('AzIoTHub_FaultOperationType', testConfiguration.operationType);
-          terminateMessage.properties.add('AzIoTHub_FaultOperationCloseReason', testConfiguration.closeReason);
-          terminateMessage.properties.add('AzIoTHub_FaultOperationDelayInSecs', testConfiguration.delayInSeconds);
-          deviceClient.sendEvent(terminateMessage, function (sendErr) {
-            debug('at the callback for the fault injection send, err is:' + sendErr);
-          });
-        } else {
+      deviceClient.open(function (err) {
+        if (err) {
+          debug('error connecting the device client: ' + err.toString());
           testCallback(err);
+        } else {
+          debug('setting up method handler for ' + methodName1);
+          setMethodHandler(methodName1, firstPayload);
+          debug('waiting 2 seconds before invoking ' + methodName1);
+          setTimeout(function () {
+            sendMethodCall(serviceClient, methodName1, firstPayload, function(err) {
```

Comment:
>sendMethodCall(serviceClient, methodName1, firstPayload, function(err) { [](start = 12, length = 72)

You might want to use the rendezvous helper on these.   I've certainly seen the response from the injection come in quite late.  You might think of this as having three paths here.  The injection path, the client method handler path, and the service sending the method path.  Just a thought.

Reply:
I'll add that if we see failures.